test/object-store/sync/session/progress_notifications.cpp

```diff
@@ -1296,4 +1303,160 @@ TEMPLATE_TEST_CASE("sync progress: upload progress", "[sync][baas][progress]", P
+
+TEST_CASE("sync progress: flx download progress", "[sync][baas][progress]") {
+    static std::optional<FLXSyncTestHarness> harness;
+    if (!harness) {
+        Schema schema{
+            {"object",
+             {{"_id", PropertyType::ObjectId, Property::IsPrimary{true}},
+              {"int", PropertyType::Int | PropertyType::Nullable},
+              {"padding", PropertyType::Data}}},
+        };
+        realm::app::FLXSyncTestHarness::ServerSchema server_schema{std::move(schema), {"int"}};
+        harness.emplace("flx_download_progress", std::move(server_schema));
+        harness->load_initial_data([](const std::shared_ptr<Realm>& realm) {
+            const size_t padding_size = 1024 * 1024;
+            auto buffer = std::make_unique<char[]>(padding_size);
+            auto table = realm->read_group().get_table("class_object");
+            for (int i = 0; i < 5; ++i) {
+                auto obj = table->create_object_with_primary_key(ObjectId::gen());
+                obj.set("int", i);
+                // ensure that each object is large enough that it'll be sent in
+                // a separate DOWNLOAD message
+                obj.set("padding", BinaryData(buffer.get(), padding_size));
+            }
+        });
+    }
+
+    SyncTestFile config = harness->make_test_file();
+
+    SECTION("async open with no subscriptions") {
+        auto task = Realm::get_synchronized_realm(config);
+        std::vector<double> estimates;
+        task->register_download_progress_notifier([&](uint64_t, uint64_t, double estimate) {
+            // Note that no locking is needed here despite this being called on
+            // a background thread as the test provides the required synchronization.
+            // If tsan complains about this, it indicates that the notifier is
+            // being called at a time that it shouldn't be and there's a bug.
+            estimates.push_back(estimate);
+        });
+        task->start().get();
+        // A download happens for the schema, but we now don't report that
+        REQUIRE(estimates.size() == 0);
+    }
+
+    SECTION("async open with initial subscriptions") {
+        config.sync_config->subscription_initializer = [](const std::shared_ptr<Realm>& realm) {
+            subscribe_to_all(*realm);
+        };
+        auto task = Realm::get_synchronized_realm(config);
+        std::vector<double> estimates;
+        task->register_download_progress_notifier([&](uint64_t, uint64_t, double estimate) {
+            // Note that no locking is needed here despite this being called on
+            // a background thread as the test provides the required synchronization.
+            // If tsan complains about this, it indicates that the notifier is
+            // being called at a time that it shouldn't be and there's a bug.
+            estimates.push_back(estimate);
+        });
+        task->start().get();
+
+        // Since we set the soft byte limit to one byte, we should have received
```

Comment:
where is the soft byte limit getting set?

Reply:
Stale comment from when I was trying to use `qbs_download_changeset_soft_max_byte_size` rather than making the objects big.